Fixed the review issues by Mattermost

manojmalik20 opened the pull request:
Fixed the SCSS structure of several components - Generic Modal, custom status, post header, status dropdown
Refactoring of code in several components
Added Menu.ItemToggleModalRedux in the set custom status option in the Status dropdown

<!-- Thank you for contributing a pull request! Here are a few tips to help you:

1. If this is your first contribution, make sure you've read the Contribution Checklist https://developers.mattermost.com/contribute/getting-started/contribution-checklist/
2. Read our blog post about "Submitting Great PRs" https://developers.mattermost.com/blog/2019-01-24-submitting-great-prs
3. Take a look at other repository specific documentation at https://developers.mattermost.com/contribute
-->

#### Summary
<!--
A description of what this pull request does.
-->

#### Ticket Link
<!--
If this pull request addresses a Help Wanted ticket, please link the relevant GitHub issue, e.g.

  Fixes https://github.com/mattermost/mattermost-server/issues/XXXXX

Otherwise, link the JIRA ticket.
-->

#### Related Pull Requests
<!--
List all PRs related to resolving a ticket. For instance, if you submitted a PR to `mattermost/mattermost-server`, please include it here.
-->
- Has server changes (please link here)
- Has redux changes (please link here)
- Has mobile changes (please link here)

#### Screenshots
<!--
If the PR includes UI changes, include screenshots/GIFs.
-->

manojmalik20 committed:
Fixed the review issues by Mattermost
Fixed the SCSS structure of several components - Generic Modal, custom status, post header, status dropdown
Refactoring of code in several components
Added Menu.ItemToggleModalRedux in the set custom status option in the Status dropdown

chetanyakan commented on components/channel_header/channel_header.js:
```diff
@@ -301,11 +301,7 @@ class ChannelHeader extends React.PureComponent {
-        } else {
-            this.setState({showCustomStatusTooltip: false});
-        }
+        this.setState({showCustomStatusTooltip: element && element.offsetWidth < element.scrollWidth});
```
nit - 
```suggestion
        const showCustomStatusTooltip = element && element.offsetWidth < element.scrollWidth;
        this.setState({showCustomStatusTooltip});
```

chetanyakan commented on components/sidebar/sidebar_channel/sidebar_channel_link/sidebar_channel_link.tsx:
```diff
@@ -101,11 +101,7 @@ export default class SidebarChannelLink extends React.PureComponent<Props, State
     // TODO: Is there a better way to do this?
```
Should we remove this TODO?

manojmalik20 replied:
I don't think so! What if there really is a better way to do this!

chetanyakan commented on components/status_dropdown/status_dropdown.jsx:
```diff
@@ -151,31 +136,30 @@ export default class StatusDropdown extends React.PureComponent {
                 <span className='d-flex'>
                     <CustomStatusEmoji
                         showTooltip={false}
+                        emojiStyle={{}}
```
Should this be an optional prop?

manojmalik20 replied:
Yes, empty object is passed here to avoid applying default styling.... If it becomes a required prop we would have to pass styling everywhere

chetanyakan replied:
Update this as discussed.

chetanyakan commented on components/widgets/menu/menu_items/menu_item_toggle_modal_redux.tsx:
```diff
@@ -15,18 +15,21 @@ type Props = {
         dialogType={dialogType}
         dialogProps={dialogProps}
-        className={extraText && 'MenuItem__with-help'}
+        className={(extraText ? 'MenuItem__with-help' : '') + (className ? ' ' + className : '')}
```
Check if we should use the classNames package here:

```js
import classNames from 'classnames';
```
```suggestion
        className={classNames({
            'MenuItem__with-help': extraText,
            className,
        })}
        
```

manojmalik20 committed:
Fixed button descendant of button error and some review fixes
Refactored the showTooltip code in all files
Changed the customStatusTextLimit
Fixed passing empty object in the emojiStyle prop in CustomStatusEmoji

manojmalik20 committed:
Made a separate component for custom status text (#24)

Modified channel header to use customStatusText component
Modified status dropdown to use customStatusText component
Modified profile popover to use customStatusText component
Modified customStatusSuggestion to use customStatusText component
